File: A17Array1/A5BinarySearch.js
let arr = [1, 2, 3, 4, 5, 6, 7, 8, 9];
let key = 6;

function binarySearch(arr, key) {
  let start = 0;
  let end = arr.length - 1;

  while (start <= end) {
    let mid = Math.floor((start + end) / 2);

    if (arr[mid] == key) {
      console.log(mid);
      return;
    }

    if (arr[mid] < key) {
      start = mid + 1;
    } else {
      end = mid - 1;
    }
  }
  console.log(-1);
  return;
}

binarySearch(arr, key);

File: A17Array1/A5BinarySearch.java
package A17Array1;

public class A5BinarySearch {
    public static int binarySearch(int arr[], int key) {
        int start = 0, end = arr.length - 1;
        while (start <= end) {
            int mid = (start + end) / 2;
            if (arr[mid] == key) {
                return mid;
            }
            if (arr[mid] < key) {
                start = mid + 1;
            } else {
                end = mid - 1;
            }
        }
        return -1;
    }

    public static void main(String args[]) {
        int arr[] = { 1, 2, 3, 4, 5, 6, 7, 8, 9 };
        int key = 6;
        System.out.println("our key is present at index " + binarySearch(arr, key));
    }
}
